File: java/scf-core/src/test/java/org/mydotey/scf/TestConfigurationSource.java
package org.mydotey.scf;

import java.util.HashMap;
import java.util.Map;

import org.mydotey.scf.AbstractConfigurationSource;
import org.mydotey.scf.ConfigurationSourceConfig;

/**
 * @author koqizhao
 *
 * May 17, 2018
 */
public class TestConfigurationSource extends AbstractConfigurationSource {

    protected Map<String, String> _properties;

    public TestConfigurationSource(ConfigurationSourceConfig config, HashMap<String, String> properties) {
        super(config);

        init();

        if (properties != null)
            properties.forEach((k, v) -> {
                if (k != null && v != null)
                    _properties.put(k, v);
            });
    }

    protected void init() {
        _properties = new HashMap<>();
    }

    @Override
    protected Object getPropertyValue(Object key) {
        return _properties.get(key);
    }

}

File: dotnet/test/MyDotey.SCFTest/TestConfigurationSource.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace MyDotey.SCF
{
    /**
     * @author koqizhao
     *
     * May 17, 2018
     */
    public class TestConfigurationSource : AbstractConfigurationSource<ConfigurationSourceConfig>
    {
        protected IDictionary<String, String> _properties;

        public TestConfigurationSource(ConfigurationSourceConfig config, Dictionary<String, String> properties)
            : base(config)
        {
            init();

            if (properties != null)
                properties.ToList().ForEach(p =>
                {
                    if (p.Key != null && p.Value != null)
                        _properties[p.Key] = p.Value;
                });
        }

        protected virtual void init()
        {
            _properties = new Dictionary<String, String>();
        }

        protected override Object getPropertyValue(Object key)
        {
            _properties.TryGetValue((String)key, out String value);
            return value;
        }
    }
}